src/providers/ldap/sdap_async_connection.c

```diff
@@ -1650,7 +1681,30 @@ static void sdap_cli_connect_done(struct tevent_req *subreq)
                                    state->service->uri,
                                    state->service->sockaddr,
                                    state->service->sockaddr_len,
-                                   state->use_tls);
+                                   state->use_tls,
+                                   state->retry_no_tls);
+
+        if (!subreq) {
+            tevent_req_error(req, ENOMEM);
+            return;
+        }
+
+        tevent_req_set_callback(subreq, sdap_cli_connect_done, req);
+        return;
+    } else if (state->retry_no_tls) {
+        DEBUG(SSSDBG_IMPORTANT_INFO, "StartTLS operation failed! Falling back to " \
+                                     "connecting with startTLS disabled. Warning: " \
+                                     "this means network communication is NOT " \
+                                     "protected with TLS.\n");
```

Comment:
Hi,

I wonder if this message should be send to the system logs as well (at least once)?

bye,
Sumit

Reply:
Added.